### PR TITLE
Catch and response the error when deleting the outdated archive

### PR DESCRIPTION
**Pull request by yma955**

### CODE REVIEW

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 1]
```diff
@@ -108,15 +106,19 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
             Map<String, String> downloadPaths = reader.readPaths( content );
             controller.downloadArtifacts( downloadPaths, content );
             Optional<File> archive = controller.generateArchive( content );
```
Comment (ruhan1) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: This is off the topic but related. The "generateArchive" should be really thread-off. This is what I mentioned early today. 
The create method don't need to wait for the generation result, it just trigger the action and tell caller "request is accepted".
To thread off, you can use Quarkus EventBus (sendAndForget(...)). ref https://quarkus.io/guides/reactive-event-bus

**Review by ruhan1 — COMMENTED** [review]
LGTM

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java (hunk 1 above)
Comment (ruhan1) [reply src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: More specifically, all the downloading/generating effort should be thread-off.

Comment (yma955) [reply src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java]: @ruhan1 Fixed in commit: https://github.com/Commonjava/indy-archive-service/pull/11/commits/7fa757dd626c0e4f0b8d28c9481ee1e5408065c7

File: src/main/java/org/commonjava/indy/service/archive/controller/ArchiveController.java [hunk 2]
```diff
@@ -114,8 +120,82 @@ public void destroy()
...
 
-    public void downloadArtifacts( final Map<String, String> downloadPaths, final HistoricalContentDTO content )
-            throws InterruptedException, ExecutionException
+    @ConsumeEvent( value = "generate-archive" )
+    public Boolean generate( HistoricalContentDTO content )
+    {
+        logger.info( "Handle generate event: {}, build config id: {}", EVENT_GENERATE_ARCHIVE,
+                     content.getBuildConfigId() );
+        Map<String, String> downloadPaths = reader.readPaths( content );
+        Optional<File> archive = Optional.empty();
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/controller/ArchiveController.java]: Just a nit-pick. This Optional.empty() is redundant.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java [hunk 3]
```diff
@@ -93,57 +94,42 @@ public Response create( final @Context UriInfo uriInfo, final @Context HttpReque
...
+            {
+                logger.error( "Archive generate event: {} is failed, build config id: {}", EVENT_GENERATE_ARCHIVE,
+                              content.getBuildConfigId() );
+            }
+        };
+        bus.request( EVENT_GENERATE_ARCHIVE, content, resultHandler );
+        return Uni.createFrom()
+                  .item( accepted().type( MediaType.TEXT_PLAIN )
+                                   .entity( "Archive created request is accepted." )
+                                   .build() );
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:123]: I'm thinking how to check if this archive is generated successfully. Maybe we need another rest endpoint to check this. So we should do:
* assign each generation task an id
* A new api to check the archive generation result by the task id.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ResponseHelper.java [hunk 4]
```diff
@@ -15,18 +15,28 @@
...
+import io.smallrye.mutiny.Uni;
+
 import javax.enterprise.context.ApplicationScoped;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.ResponseBuilder;
 
+import static javax.ws.rs.core.Response.serverError;
+
 @ApplicationScoped
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/jaxrs/ResponseHelper.java]: So this @ApplicationScoped is not needed as methods are all static.

**Review by ligangty — COMMENTED** [review]
@yma88 Totally looks good, but still something need to consider.For that archive generation result checking issue, if you don't want to add extra commits to add, you can open a new jira issue for tracking and add that later.

File: src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java (hunk 3 above)
Comment (yma955) [reply src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java:123]: @ligangty Yeah, I'm also thinking about this, currently users could check the log output to confirm the generation result, taking in the handler of AsyncResult. Another point is the sidecar will check the archive availability before per download, if the archive is empty or content compressed is not available, the sidecar will doProxy as simple.

Sure I agree that we could add the REST of checking result status to make users convenient to notice the generation result, I'll file a story for this. Thanks!

**Comment by yma955** [comment]
push the amending commit, and for generation result checking, file story https://projects.engineering.redhat.com/browse/MMENG-1535.

**Review by ligangty — COMMENTED** [review]
LGTM

**Pull request merged**